packages/chai-openapi-response-validator/test/assertions/satisfyApiSpec/basePathVariations.test.js

```diff
@@ -0,0 +1,159 @@
+/** *****************************************************************************
+ * Copyright 2019 IBM Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ****************************************************************************** */
+
+const chai = require('chai');
+const path = require('path');
+
+const chaiResponseValidator = require('../../..');
+
+const dirContainingApiSpec = path.resolve(
+  '../../commonTestResources/exampleOpenApiFiles/valid/basePathVariations',
+);
+const { expect, AssertionError } = chai;
+
+describe('Using OpenAPI 2 specs that vary on whether they define basePath', () => {
```

Comment:
I understand that by 'not defined' you mean the user hasn't provided a basePath property in their spec, however a name more consistent with our [existing OpenAPI 3 server tests](https://github.com/RuntimeTools/OpenAPIValidators/blob/master/packages/chai-openapi-response-validator/test/assertions/satisfyApiSpec/specsDefineServersDifferently.test.js#L28) would be:
```suggestion
describe('Using OpenAPI 2 specs that define basePath differently', () => {
```

(And rename `basePathNotDefined.yml` to `noBasePathProperty.yml`)

Naming is one of most subjective things about this package, but we can at least be consistent. 

(For some reason, for the OpenAPI 3 `specsDefineServersDifferently.test.js` I decided that a spec not providing a `server` property would still have an "implicitly defined" `server`, so to speak)

p.s. The official OpenAPI 3 specification seems to use "provided":
> If the servers property is not provided, or is an empty array, the default value would be a Server Object with a url value of /.

i.e. a user may or may not provide a `basePath`. 

If you agree, I can change "defined" to "provided" in a future PR (rather than bloat this PR)

Reply:
Hmm, for this one I would just differ to your best judgement. When we were discussing error messages, I felt like coming to this package as a user gave me a useful POV. But you know more about this package and OpenApi, so I think you know best re naming conventions. I do agree that it should be a separate PR, though.

As for the changes suggested, they were included in [Changed 'urlErrorCodes' to 'validationErrorCodes' ](https://github.com/RuntimeTools/OpenAPIValidators/pull/172/commits/35079f970d8c26ec41077ad2d3850bdddad48f5a)

Reply:
Cool, yea I've been experimentally converting this codebase into TypeScript, and have found our naming to be sometimes inconsistent with the types and the official OpenAPI Specification. So I will tidy the names up in a future PR